packages/core/src/compiler/Style.ts

```diff
@@ -2516,9 +2512,15 @@ const isFieldOrAccessPath = (p: Path): boolean => {
 // This also samples varying access paths, e.g.
 // Circle { center : (1.1, ?) ... } <// the latter is an access path that gets initialized here
 // NOTE: Mutates translation
-const initFields = (varyingPaths: Path[], tr: Translation): Translation => {
-  const varyingFields = varyingPaths.filter(isFieldOrAccessPath);
-  const sampledVals = randFloats(varyingFields.length, canvasXRange);
+const initFieldsAndAccessPaths = (
+  varyingPaths: Path[],
+  tr: Translation
+): Translation => {
+  const varyingFieldsAndAccessPaths = varyingPaths.filter(isFieldOrAccessPath);
+  const sampledVals = randFloats(
```

Comment:
For a temporary solution, this seems to be fine. Ideally, I'd prefer if we have a way to operate on `Shape[]` instead of varying paths directly. This is somewhat connected to #513. The main problem we have seems to be: if we only look at shapes, we have no idea if a property came from a field, e.g.
```
Set X {
  val = ?
  X.icon = Circle { r: val }
}
```
In the translation, `X.icon.r` is a `LocalVar`. In `state.varyingPaths`, `val` is some local field path. So when dragging a shape, there's no obvious way to "trace" the origin of `X.icon.r` and back propagate the changes. 

I guess another possible solution is to maintain some data structure that points from shape properties to varying paths. We can (1) sample shapes only, pull prop  values out, and insert them into `varyingValues` and (2) update shapes on front-end updates (e.g. drag) and do the same thing to get the next `varyingValues`.

Reply:
Hmm. I will give this more careful thought and get back to you tonight.

Reply:
> So when dragging a shape, there's no obvious way to "trace" the origin of `X.icon.r` and back propagate the changes.

Why not look in the translation for `X.icon.r`? That would give you `X.val` which you can then lookup in `varyingPaths`/`varyingValues` to confirm it's a varying value and update it with the results of the drag. Not sure if I'm missing something about this method.

Reply:
Here's a screenshot of the example above:

![image](https://user-images.githubusercontent.com/11740102/110576030-f9d61380-812d-11eb-9691-a1a739ab4eaf.jpeg)

Reply:
Ok, this is indeed a bug in the Style compiler. All the local vars should be substituted out, ideally. Thanks for reporting it! Issue tracked here, will aim to fix today. https://github.com/penrose/penrose/issues/523